packages/solid/src/reactive/signal.ts

```diff
@@ -752,14 +752,13 @@ export function createSelector<T, U>(
   const node = createComputation(
     (p: T | undefined) => {
       const v = source();
-      for (const [key, val] of subs.entries())
-        if (fn(key, v) !== fn(key, p!)) {
-          for (const c of val.values()) {
-            c.state = STALE;
-            if (c.pure) Updates!.push(c);
-            else Effects!.push(c);
-          }
+      for (const [key, val] of subs.entries()) {
+        if (fn(key, v) === fn(key, p!)) continue
+        for (const c of val.values()) {
+          c.state = STALE;
+          c.pure ? Updates!.push(c) : Effects!.push(c)
```

Comment:
I would argue that this is less readable and debuggable. Also, the minification will result in the same output anyways.